### PR TITLE
Internal: Container fixes [ED-5605]

### PR DESCRIPTION
**Pull request by StyleShit**
~Depends on #16964~

### CODE REVIEW

File: assets/dev/js/editor/document/hooks/ui/document/elements/create/index.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export { MoveResizeableHandle } from './move-resizeable-handle';
```
Comment (StyleShit) [inline assets/dev/js/editor/document/hooks/ui/document/elements/create/index.js:1]: @matipojo 
This PR depends on #16964

File: assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js [hunk 2]
```diff
@@ -17,8 +17,14 @@ export class SetDirectionMode extends After {
...
 
 	apply( args ) {
+		SetDirectionMode.set( args.container?.view );
+	}
+
+	static set( view ) {
 		// Get the direction mode from the view & set the state accordingly.
-		const direction = args.container?.view?.getCurrentUiStates?.().directionMode;
+		// If it's a Widget, use its parent to determine the direction.
+		const containingView = ( 'widget' === view?.model.get( 'elType' ) ) ? view._parent : view,
```
Comment (Nevoss) [inline assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js:26]: What if the child is a container? maybe 'parentDirectionMode'

File: assets/dev/js/editor/utils/container-helper.js [hunk 3]
```diff
@@ -84,10 +84,16 @@ export class ContainerHelper {
 			66: '66.6666',
 		};
 
+		const sizesSum = sizes.reduce( ( sum, size ) => {
+			return sum + parseInt( size );
+		}, 0 );
+
+		const shouldWrap = ( sizesSum > 100 );
+
```
Comment (Nevoss) [inline assets/dev/js/editor/utils/container-helper.js:92]: I think it should be always wrap @matipojo @shilo-ey

File: assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js [hunk 4]
```diff
@@ -17,8 +17,14 @@ export class SetDirectionMode extends After {
 	}
 
 	apply( args ) {
+		SetDirectionMode.set( args.container?.view );
```
Comment (matipojo) [inline assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js:20]: The `settings` command requires a `container` so, it must exist. Also, it's better to use the container itself.

```suggestion
		SetDirectionMode.set( args.container );
```

File: assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js (hunk 2 above)
Comment (matipojo) [reply assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js:26]:
```suggestion
		if ( 'widget' === container.type ) {
			container = container.parent;
		}
```

File: assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js [hunk 5]
```diff
@@ -17,8 +17,14 @@ export class SetDirectionMode extends After {
...
+	}
+
+	static set( view ) {
```
Comment (matipojo) [inline assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js:23]:
```suggestion
	static set( container ) {
```

File: assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js [hunk 6]
```diff
@@ -17,8 +17,14 @@ export class SetDirectionMode extends After {
...
 		// Get the direction mode from the view & set the state accordingly.
-		const direction = args.container?.view?.getCurrentUiStates?.().directionMode;
+		// If it's a Widget, use its parent to determine the direction.
+		const containingView = ( 'widget' === view?.model.get( 'elType' ) ) ? view._parent : view,
+			direction = containingView?.getCurrentUiStates?.().directionMode;
```
Comment (matipojo) [inline assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js:27]: Try use the `renderer` that refers to the element with a view
```suggestion
			direction = container.renderer.view.getCurrentUiStates().directionMode;
```

File: assets/dev/js/editor/regions/panel/pages/editor/component.js [hunk 7]
```diff
@@ -87,16 +88,7 @@ export default class Component extends ComponentBase {
...
-		const container = elementor.getCurrentElement().container,
-			direction = container?.view.getCurrentUiStates?.().directionMode;
-
-		if ( direction ) {
-			$e.uiStates.set( 'document/direction-mode', direction );
-			return;
-		}
-
-		$e.uiStates.remove( 'document/direction-mode' );
+		SetDirectionMode.set( elementor.getCurrentElement().container?.view );
```
Comment (matipojo) [inline assets/dev/js/editor/regions/panel/pages/editor/component.js:91]:
```suggestion
		SetDirectionMode.set( elementor.getCurrentElement().container );
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/regions/panel/pages/editor/component.js (hunk 7 above)
Comment (StyleShit) [reply assets/dev/js/editor/regions/panel/pages/editor/component.js:91]: Done

File: assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js (hunk 2 above)
Comment (StyleShit) [reply assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js]: Done

File: assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js (hunk 4 above)
Comment (StyleShit) [reply assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js]: Done

File: assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js (hunk 5 above)
Comment (StyleShit) [reply assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js]: Done

File: assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js (hunk 6 above)
Comment (StyleShit) [reply assets/dev/js/editor/document/hooks/ui/settings/set-direction-mode.js]: Done

**Review by matipojo — APPROVED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]
Please check the case that the `renderer` does not exist, and try to make it always exist.
(see QUnit error)

**Comment by StyleShit** [comment]
@matipojo 
That wasn't the problem, but fixed.

**Pull request merged**